File: Algorithms/src/3163. String Compression III/Code/solution.go
func compressedString(word string) string {
    var comp string
    count := 1
    
    for i := 1; i <= len(word); i++ {
        if i == len(word) || word[i] != word[i-1] || count == 9 {
            comp += fmt.Sprintf("%d%c", count, word[i-1])
            count = 1
        } else {
            count++
        }
    }
    
    return comp
}

File: Algorithms/src/3163. String Compression III/Code/solution.js
/**
 * @param {string} word
 * @return {string}
 */
var compressedString = function (word) {
  let comp = "";
  let count = 1;

  for (let i = 1; i <= word.length; i++) {
    if (i === word.length || word[i] !== word[i - 1] || count === 9) {
      comp += count + word[i - 1];
      count = 1;
    } else {
      count++;
    }
  }

  return comp;
};
